lib/sidekiq/grouping/flusher.rb

```diff
@@ -3,18 +3,38 @@ def flush
     batches = Sidekiq::Grouping::Batch.all.map do |batch|
       batch if batch.could_flush?
     end
-    batches.compact!
-    flush_concrete(batches)
+    flush_batches(batches)
+  end
+
+  def force_flush_for_test!
+    if defined?(::Rails) && Rails.respond_to?(:env) && !Rails.env.test?
+      Sidekiq::Grouping.logger.warn(
+        "**************************************************"
+      )
+      Sidekiq::Grouping.logger.warn([
+        "⛔️ force_flush_for_test! for testing API, ",
+        "but this is not the test environment."
+      ].join)
+      Sidekiq::Grouping.logger.warn(
+        "**************************************************"
+      )
+    end
+    flush_batches(Sidekiq::Grouping::Batch.all)
   end
 
   private
 
+  def flush_batches(batches)
+    batches.compact!
+    flush_concrete(batches)
+  end
+
   def flush_concrete(batches)
     return if batches.empty?
     names = batches.map { |batch| "#{batch.worker_class} in #{batch.queue}" }
     Sidekiq::Grouping.logger.info(
       "[Sidekiq::Grouping] Trying to flush batched queues: #{names.join(',')}"
-    )
+    ) unless defined?(::Rails) && Rails.respond_to?(:env) && Rails.env.test?
```

Comment:
What do you think, if we extract this condition into separate private method like ```#test?```. Moreover, it might be good to have global setting ```Sidekiq::Grouping.test_mode``` because this gem might be used along with other frameworks and it would be good option to explicitly set this variable somewhere in spec_helper for non-Rails environments.